### PR TITLE
Add mechanism to set default status(on/off) of plugin

### PR DESCRIPTION
**Pull request by ascrutae**

### CODE REVIEW

**Comment by ascrutae** [comment]
Relate #245

File: apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginDefine.java [hunk 1]
```diff
@@ -30,16 +36,31 @@ public static PluginDefine build(String define) throws IllegalPluginDefineExcept
             throw new IllegalPluginDefineException(define);
         }
 
-        return new PluginDefine(pluginDefine[0], pluginDefine[1]);
+        String pluginName = pluginDefine[0];
+        String defineClass = pluginDefine[1];
+        if (pluginName.toUpperCase().startsWith("[OFF]")) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginDefine.java]: Set "[OFF]" as a static field.

File: apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginDefine.java [hunk 2]
```diff
@@ -30,16 +36,31 @@ public static PluginDefine build(String define) throws IllegalPluginDefineExcept
...
+            return new PluginDefine(pluginName.substring(5), defineClass, State.OFF);
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginDefine.java]: `5` should be replaced by the length of new field. ^^^

**Comment by wu-sheng** [comment]
Ref #242 , and cc @pengys5 You must be interested in this feature. Your Kafka plugin must use it.

**Comment by ascrutae** [comment]
I had added test case for `Plugin Define`.

**Comment by wu-sheng** [comment]
cc @pengys5 This feature has been merged. Kafka plugin should follow this.

**Pull request merged**